src/main/java/uk/gov/hmcts/opal/service/LegacyService.java

```diff
@@ -27,23 +28,35 @@ protected LegacyService(String gatewayUrl, RestClient restTemplate) {
     protected abstract Logger getLog();
 
     public <T> T extractResponse(ResponseEntity<String> responseEntity, Class<T> clzz) {
-        if (responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
-            String rawJson = responseEntity.getBody();
-            getLog().info("extractResponse: Raw JSON response: {}", rawJson);
-
-            try {
-                ObjectMapper objectMapper = ToJsonString.getObjectMapper();
-                JsonNode root = objectMapper.readTree(rawJson);
-
-                return objectMapper.treeToValue(root, clzz);
-
-            } catch (Exception e) {
-                getLog().error("extractResponse: Error deserializing response: {}", e.getMessage(), e);
+        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+            if (responseEntity.getBody() != null) {
+
+                String rawJson = responseEntity.getBody();
+                getLog().info("extractResponse: Raw JSON response: {}", rawJson);
+
+                try {
+                    ObjectMapper objectMapper = ToJsonString.getObjectMapper();
+                    JsonNode root = objectMapper.readTree(rawJson);
+
+                    return objectMapper.treeToValue(root, clzz);
+
+                } catch (Exception e) {
+                    getLog().error("extractResponse: Error deserializing response: {}", e.getMessage(), e);
+                    throw new LegacyGatewayResponseException(e);
+                }
```

Comment:
I think this can be extracted as a utility method along with the code in line 81 from this file, in a separate class i.e. `JsonMapper` or something similar. 
```suggestion
@Componenet
class JsonMapper {
    private static final String SERIALISATION_ERROR_MESSAGE = "Failed to serialize '%s' to JSON";
    private static final String DESERIALIZATION_ERROR_MESSAGE = "Failed to deserialize '%s' from JSON";

private final ObjectMapper objectMapper;

    public JsonMapper(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }
    
   public <T> T fromJson(String value, Class<T> clazz) {
        try {
            return objectMapper.readValue(value, clazz);
        } catch (IOException e) {
            throw new SomeApplicationException(
                String.format(DESERIALIZATION_ERROR_MESSAGE, clazz.getSimpleName()), e
            );
        }
    }

public String toJson(Object input) {
        try {
            return objectMapper.writeValueAsString(input);
        } catch (JsonProcessingException e) {
            throw new SomeApplicationException(
                String.format(SERIALISATION_ERROR_MESSAGE, input.getClass().getSimpleName()), e
            );
        }
    }
}                
```

Reply:
Would you like to do that after you've approved this PR?